pkg/receive/handler_otlp.go

```diff
@@ -0,0 +1,171 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package receive
+
+import (
+	"context"
+	"net/http"
+	"strconv"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/pkg/errors"
+	"github.com/prometheus/prometheus/storage/remote"
+	"github.com/thanos-io/thanos/pkg/receive/otlptranslator"
+	tprompb "github.com/thanos-io/thanos/pkg/store/storepb/prompb"
+	"github.com/thanos-io/thanos/pkg/tenancy"
+	"github.com/thanos-io/thanos/pkg/tracing"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+)
+
+func (h *Handler) receiveOTLPHTTP(w http.ResponseWriter, r *http.Request) {
+	var err error
+	span, ctx := tracing.StartSpan(r.Context(), "receiveOTLPHTTP")
+	span.SetTag("receiver.mode", string(h.receiverMode))
+	defer span.Finish()
+
+	tenant, err := tenancy.GetTenantFromHTTP(r, h.options.TenantHeader, h.options.DefaultTenantID, h.options.TenantField)
```

Comment:
It seems like the whole section for handling tenancy limits is 1:1 with code in the native handler. Maybe we could have a small common function to handle it in both places?

Reply:
I think it makes sense, but maybe address it on another PR, I'd like to avoid change code from main handler on this PR.

wdyt @matej-g ?

Reply:
Sure let's do that!